packages/dd-trace/src/plugins/util/user-provided-git.js

```diff
@@ -39,6 +41,39 @@ function filterSensitiveInfoFromRepository (repositoryUrl) {
   }
 }
 
+function validateGitRepositoryUrl (repoUrl) {
+  if (repoUrl) {
+    const validUrl = /((http|git|ssh|http(s)|file|\/?)|(git@[\w.]+))(:(\/\/)?)([\w.@:/\-~]+)(\.git)(\/)?/.test(repoUrl)
+    if (validUrl) {
+      return true
+    }
+  }
+  return false
+}
+
+function validateGitCommitSha (gitCommitSha) {
+  if (gitCommitSha) {
+    const isValidSha1 = /^[0-9a-f]{40}$/.test(gitCommitSha)
+    const isValidSha256 = /^[0-9a-f]{64}$/.test(gitCommitSha)
+    if (isValidSha1 || isValidSha256) {
+      return true
+    }
+  }
+  return false
+}
+
+function validateGitMetadata (metadata) {
+  const size = Object.keys(metadata).length
+  if (size > 0) {
+    if (!validateGitRepositoryUrl(metadata[GIT_REPOSITORY_URL])) {
+      log.error('DD_GIT_COMMIT_SHA must be a full-length git SHA')
+    }
+    if (!validateGitCommitSha(metadata[GIT_COMMIT_SHA])) {
+      log.error('DD_GIT_REPOSITORY_URL must be a valid URL')
+    }
+  }
+}
```

Comment:
logging an error is a good idea, but we want to end up doing is removing the invalid values. We could do something like what we do with `removeEmptyValues`
```suggestion
function removeInvalidGitMetadata (metadata) {
  return Object.keys(metadata).reduce((filteredTags, tag) => {
    if (tag === GIT_REPOSITORY_URL) {
      if (!validateGitRepositoryUrl(metadata[GIT_REPOSITORY_URL])) {
        log.error('DD_GIT_COMMIT_SHA must be a full-length git SHA')
        return filteredTags
      }
    }
    if (tag === GIT_COMMIT_SHA) {
      if (!validateGitCommitSha(metadata[GIT_COMMIT_SHA])) {
        log.error('DD_GIT_REPOSITORY_URL must be a valid URL')
        return filteredTags
      }
    }
    filteredTags[tag] = metadata[tag]
    return filteredTags
  }, {})
}
```